### PR TITLE
openstack-ardana: add option to use salt API to create SES resources

### PR DESCRIPTION
**Pull request by flaviodsr**
Add support for ardana SES integration v2, which uses the salt-api to
configure ceph for openstack.

It is possible to configure which version of ardana SES integration to
use by setting `ardana_ses_integration_version` which defaults to 2.

### CODE REVIEW

**Comment by flaviodsr** [comment]
Tested in:

https://ci.suse.de/blue/organizations/jenkins/openstack-ardana/detail/openstack-ardana/2780/pipeline/126

**Comment by hemna** [comment]
For what it's worth, the shipping version of deepsea still has a bug that makes the openstack.integrate sometimes output unicode garbage.    The bug has been fixed, but it's not out in the wild yet.

https://github.com/SUSE/DeepSea/issues/1495

File: scripts/jenkins/ardana/ansible/roles/ses_ardana_integration/tasks/ses_configure_v1.yml [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+    content: "{{ item.content | b64decode }}"
+    mode: 0644
+  when: "item.item.name != 'client.rgw.ses-' ~ ardana_env"
+  loop: "{{ hostvars['ses-' + ses_cluster_id].ceph_files.results }}"
+  loop_control:
+    label: "{{ item.item.name }}"
+
+- name: Generate ses-config.yml
+  template:
+    src: ses-config.yml.j2
```
Comment (hemna) [inline scripts/jenkins/ardana/ansible/roles/ses_ardana_integration/tasks/ses_configure_v1.yml]: probably should name this template ses-config-v1.yml.j2

File: scripts/jenkins/ardana/ansible/roles/ses_ardana_integration/defaults/main.yml [hunk 2]
```diff
@@ -15,6 +15,7 @@
 #
 ---
 
+ardana_ses_integration_version: 2
```
Comment (stefannica) [inline scripts/jenkins/ardana/ansible/roles/ses_ardana_integration/defaults/main.yml]: You already have this in `group_vars/all/all.yml`

File: scripts/jenkins/ses/ansible/bootstrap-ses-node.yml [hunk 3]
```diff
@@ -49,6 +49,7 @@
       zypper_repository:
         name: "{{ item.name }}"
         repo: "{{ item.repo }}"
+        auto_import_keys: "{{ item.import_key | default(False) }}"
```
Comment (stefannica) [inline scripts/jenkins/ses/ansible/bootstrap-ses-node.yml]: This doesn't seem to be actively used.

File: scripts/jenkins/ses/ansible/bootstrap-ses-node.yml [hunk 4]
```diff
@@ -14,7 +14,7 @@
 # under the License.
 #
 ---
-  
+
```
Comment (stefannica) [inline scripts/jenkins/ses/ansible/bootstrap-ses-node.yml]: no reason to include this file in your PR now :)

**Comment by flaviodsr** [comment]
With CentOS 7.5 computes:

https://ci.suse.de/blue/organizations/jenkins/openstack-ardana/detail/openstack-ardana/3425/pipeline/

File: scripts/jenkins/ardana/ansible/roles/ses_ardana_integration/tasks/ses_configure_v1.yml (hunk 1 above)
Comment (JanZerebecki) [reply scripts/jenkins/ardana/ansible/roles/ses_ardana_integration/tasks/ses_configure_v1.yml]: Yes, please follow up in a new PR.

**Pull request merged**